lib/screens/auth/email_sign_in_screen.dart

```diff
@@ -185,8 +190,12 @@ class _EmailPasswordFormState extends State<_EmailPasswordForm> {
                           : _onForgotPasswordClick,
                     ),
                   ElevatedButton(
-                    child: const Text('Sign In'),
-                    onPressed: !isSignInButtonEnabled ? null : _onSignInClick,
+                    child: Text(signInButtonText),
+                    onPressed: (){setState(() {
+                      Text(signInButtonText);
+                    });
+                      !isSignInButtonEnabled ? null : _onSignInClick();} ,
+                    
```

Comment:
Also I think you meant to do this:
```suggestion
                    child: Text(signInButtonText),
                    onPressed: (){setState(() {
                        !isSignInButtonEnabled ? null : _onSignInClick();} ,
                    });
                    
```

Reply:
> Also I think you meant to do this:

Even better is to avoid defining the `onPressed` if it's not going to do anything

```dart
onPressed: isSignInButtonEnabled ? _onSignInClick : null
```

Also notice I didn't create a function to call a void function, I can just pass the function name as the value of `onPressed`, both for performance and simplicity reasons

Reply:
Even better is to avoid defining the onPressed if it's not going to do anything

onPressed: isSignInButtonEnabled ? _onSignInClick : null
Also notice I didn't create a function to call a void function, I can just pass the function name as the value of onPressed, both for performance and simplicity reasons

Reply:
Its probably doing something if it was already there?

Reply:
I'm not I understand what you mean

Reply:
you're right. sorry my suggestion is wrong.

Yea lets keep my old code:
```dart
                    onPressed: !isSignInButtonEnabled ? null : _onSignInClick,
```

Reply:
The thing I suggested is different than your old code. It sets the value of onPressed conditionally rather than conditionally doing logic inside an anonymous function. I think the way I suggested is shorter, easier to read, and more performant

Reply:
I wasn't too concerned about the onPressed function. Personally, I think it makes sense to not check the negated version of the boolean if the logic can be done that way, so I'd fancy Aashir's revision.

Reply:
Also, uhm @aashir-khan I tried just passing in the name of the function as the value, it gives an error saying we're trying to use the value of a void type function
<img width="892" alt="Screen Shot 2021-07-10 at 17 43 21" src="https://user-images.githubusercontent.com/44068314/125176992-54e44980-e1a6-11eb-9b9d-dd83497d8cbb.png">
if your concern was meant for the setState function being called in your earlier comment. Unless I'm doing it wrong, but this from what I can tell is back to the original version of how it was from v4.0

Reply:
You shouldn't be calling `_onSignInClick`, you just give the function name.

The value of `onPressed` should be a void returning and no parameter taking function if defined
![image](https://user-images.githubusercontent.com/25349581/125177491-27999a80-e1aa-11eb-98f5-b0b7e9f0f0a6.png)

which is exactly what is `_onSignInClick` is

Reply:
If you call the function, you are providing a value of type `void` (the return value of `_onSignInClick` to `onPressed`, but `onPressed` is expecting a function, not a value

Reply:
@aashir how's your suggestion different than my old code except that I had negated the boolean? I was passing the function as is too.

Reply:
![image](https://user-images.githubusercontent.com/25349581/125177747-63356400-e1ac-11eb-9d67-dab69259a46e.png)

I'm assuming you "old code" as in what is in the suggestion (but same is applicable to how it was set up before this PR was introduced). This is an anonymous function. In the code you mentioned, `onPressed` is always defined no matter what the value of `isSignInButtonEnabled`, and what happens is that the function always gets called, but depending on the value of `isSignInButtonEnabled` either something happens or nothing happens

What I did was to say that if `isSignInButtonEnabled` is defined, then `onPressed` can be assigned to `_onSignInClick` else `onPressed === null`

Reply:
I just realized that your suggestion is based on the PR changes, and what you probably actually meant by `old code` is how it was before the PR was introduced ([this](isSignInButtonEnabled)). If so, yea in that case nvm, you had basically same thing as what I suggested except the negation being used

Reply:
yea my suggestion was wrong, by old code I meant what's already there on branch `v4.0`:

https://github.com/owais635/planner_flutter/blob/b1342c834d15f97d857cc10a94231fa936fe1359/lib/screens/auth/email_sign_in_screen.dart#L187-L190

Reply:
aaaaaah I just looked at it more closely, I'll revert it to the v4.0 version.